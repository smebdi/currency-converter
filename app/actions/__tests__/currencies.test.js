import { getInitialConversion, changeCurrencyAmount } from '../currencies';

describe('getInitialConversion', () => {
    it('creates a properly formatted action', () => {
        const expected = { type: 'GET_INITIAL_CONVERSION' }
        const actual = getInitialConversion();
        expect(expected).toEqual(actual);
    });
});

describe('getInitialConversionSnapshot', () => {
    expect(getInitialConversion()).toMatchSnapshot();
});

describe('changeCurrencyAmount', () => {
    it('creates an action with an amount of 100', () => {
        const expected = { type: 'CHANGE_CURRENCY_AMOUNT', amount: 100 }
        const actual = changeCurrencyAmount(100);
        expect(expected).toEqual(actual);
    });

    it('converts a string value to a float', () => {
        const expected = { type: 'CHANGE_CURRENCY_AMOUNT', amount: 100 }
        const actual = changeCurrencyAmount('100');
        expect(expected).toEqual(actual);
    });
});

describe('changeCurrencyAmountSnapshot', () => {
    it('creates an action with an amount of 100', () => {
        expect(changeCurrencyAmount(100)).toMatchSnapshot();
    });

    it('converts a string value to a float', () => {
        expect(changeCurrencyAmount('100')).toMatchSnapshot();
    });
});
