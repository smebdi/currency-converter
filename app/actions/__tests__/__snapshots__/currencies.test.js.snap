// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "type": "GET_INITIAL_CONVERSION",
}
`;

exports[`changeCurrencyAmountSnapshot converts a string value to a float 1`] = `
Object {
  "amount": 100,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;

exports[`changeCurrencyAmountSnapshot creates an action with an amount of 100 1`] = `
Object {
  "amount": 100,
  "type": "CHANGE_CURRENCY_AMOUNT",
}
`;
